### ISSUE TITLE
Error when round starts

### ISSUE THREAD

**Issue by Aviriix** [caseif/TTT#29]
Hello,

First I wanted to thank you for getting a fix out so fast for the previous issue I created, thanks a lot :).

But, onto the bug report. I re-created my arena, which worked really well. But after the preparation round, noone got their items and the console gave me this error:

```
[00:11:42] [Server thread/ERROR]: [Steel] Failed to dispatch event class net.caseif.flint.common.event.round.CommonRoundChangeLifecycleStageEvent to public void net.caseif.ttt.listeners.MinigameListener.onRoundPrepare(net.caseif.flint.event.round.RoundChangeLifecycleStageEvent)
[00:11:42] [Server thread/WARN]: java.lang.IllegalStateException: Unregistered scoreboard component
[00:11:42] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scoreboard.CraftTeam.checkState(CraftTeam.java:217)
[00:11:42] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scoreboard.CraftTeam.addEntry(CraftTeam.java:142)
[00:11:42] [Server thread/WARN]:    at net.caseif.ttt.scoreboard.ScoreboardManager.update(ScoreboardManager.java:193)
[00:11:42] [Server thread/WARN]:    at net.caseif.ttt.listeners.MinigameListener.startRound(MinigameListener.java:148)
[00:11:42] [Server thread/WARN]:    at net.caseif.ttt.listeners.MinigameListener.onRoundPrepare(MinigameListener.java:140)
[00:11:42] [Server thread/WARN]:    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[00:11:42] [Server thread/WARN]:    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[00:11:42] [Server thread/WARN]:    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[00:11:42] [Server thread/WARN]:    at java.lang.reflect.Method.invoke(Method.java:497)
[00:11:42] [Server thread/WARN]:    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
[00:11:42] [Server thread/WARN]:    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
[00:11:42] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
[00:11:42] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
[00:11:42] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
[00:11:42] [Server thread/WARN]:    at net.caseif.flint.common.round.CommonRound.setLifecycleStage(CommonRound.java:245)
[00:11:42] [Server thread/WARN]:    at net.caseif.flint.common.round.CommonRound.nextLifecycleStage(CommonRound.java:289)
[00:11:42] [Server thread/WARN]:    at net.caseif.flint.steel.round.RoundWorker.handleTick(RoundWorker.java:76)
[00:11:42] [Server thread/WARN]:    at net.caseif.flint.steel.round.RoundWorker.run(RoundWorker.java:58)
[00:11:42] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:71)
[00:11:42] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:350)
[00:11:42] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:722)
[00:11:42] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:374)
[00:11:42] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:653)
[00:11:42] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:556)
[00:11:42] [Server thread/WARN]:    at java.lang.Thread.run(Thread.java:745)
```

This happened at the start of the round, I'm running the latest spigot version and Java 8.

**Comment by caseif** [comment]
Have you tried restarting the server? I've had this happen on my testing server before, but I've never been able to reliably replicate it. Usually a restart fixes it.

**Comment by Aviriix** [comment]
I just restarted the server, it still gave an error. It seems to be the same one, but I'll paste it again just incase.

```
[00:30:43] [Server thread/ERROR]: [Steel] Failed to dispatch event class net.caseif.flint.common.event.round.CommonRoundChangeLifecycleStageEvent to public void net.caseif.ttt.listeners.MinigameListener.onRoundPrepare(net.caseif.flint.event.round.RoundChangeLifecycleStageEvent)
[00:30:43] [Server thread/WARN]: java.lang.IllegalStateException: Unregistered scoreboard component
[00:30:43] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scoreboard.CraftTeam.checkState(CraftTeam.java:217)
[00:30:43] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scoreboard.CraftTeam.addEntry(CraftTeam.java:142)
[00:30:43] [Server thread/WARN]:    at net.caseif.ttt.scoreboard.ScoreboardManager.update(ScoreboardManager.java:193)
[00:30:43] [Server thread/WARN]:    at net.caseif.ttt.listeners.MinigameListener.startRound(MinigameListener.java:148)
[00:30:43] [Server thread/WARN]:    at net.caseif.ttt.listeners.MinigameListener.onRoundPrepare(MinigameListener.java:140)
[00:30:43] [Server thread/WARN]:    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[00:30:43] [Server thread/WARN]:    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[00:30:43] [Server thread/WARN]:    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[00:30:43] [Server thread/WARN]:    at java.lang.reflect.Method.invoke(Method.java:497)
[00:30:43] [Server thread/WARN]:    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
[00:30:43] [Server thread/WARN]:    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
[00:30:43] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
[00:30:43] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
[00:30:43] [Server thread/WARN]:    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
[00:30:43] [Server thread/WARN]:    at net.caseif.flint.common.round.CommonRound.setLifecycleStage(CommonRound.java:245)
[00:30:43] [Server thread/WARN]:    at net.caseif.flint.common.round.CommonRound.nextLifecycleStage(CommonRound.java:289)
[00:30:43] [Server thread/WARN]:    at net.caseif.flint.steel.round.RoundWorker.handleTick(RoundWorker.java:76)
[00:30:43] [Server thread/WARN]:    at net.caseif.flint.steel.round.RoundWorker.run(RoundWorker.java:58)
[00:30:43] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:71)
[00:30:43] [Server thread/WARN]:    at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:350)
[00:30:43] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:722)
[00:30:43] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:374)
[00:30:43] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:653)
[00:30:43] [Server thread/WARN]:    at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:556)
[00:30:43] [Server thread/WARN]:    at java.lang.Thread.run(Thread.java:745)
```

**Comment by caseif** [comment]
Hm, I'll need to look into this further. I'm feeling a little burned out at the moment but I'll try to take a look tomorrow and get another hotfix out ASAP. Thanks for reporting these by the way, it helps me out tremendously.

**Comment by Aviriix** [comment]
No worries, thanks for creating these for free and for the support :), I've been meaning to ask by the way, I've seen the 'Donation maybe' section on your BukkitDev page, but I can't seem to find a button or info to donate, do you still accept donations?

**Comment by caseif** [comment]
There's a link at the top right corner of the page, although it sort of blends into the navbar. I've been meaning myself to add a more visible link to the description.

Edit: Just added a fancy yellow one.

**Comment by Aviriix** [comment]
Alright, atm my server is still in a closed alpha, but when it opens up I'll make sure to send a donation :)

**Comment by caseif** [comment]
Did you restart the server via the `/reload` command, or via a full restart?

**Comment by Aviriix** [comment]
I did a full restart

**Comment by caseif** [comment]
I can't seem to replicate this. Can you provide the output of the `/pl` command?

**Comment by Aviriix** [comment]
-- image removed --

**Comment by caseif** [comment]
That's quite a list... Would you mind shooting me an email at me@caseif.net so we can try to get this sorted a bit more quickly? :)

**Comment by Aviriix** [comment]
Ofcourse, you should receive one now.

**Comment by caseif** [comment]
For the record, this seems to be an incompatibility with [RankPrefix+](http://dev.bukkit.org/bukkit-plugins/rankprefix-plus/).

**Comment by caseif** [comment]
Unfortunately, a fix would significantly complicate the codebase for the sake of compatibility with one plugin, and would likely introduce other bugs later down the line.

**Comment by caseif** [comment]
Fixed as a side-effect of 1678dd1.
